src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -0,0 +1,257 @@
+    summary_line(io, "#systems", nsystems(semi))
+    for i = 1:nsystems(semi)
+      summary_line(io, "system", i)
+      mesh, equations, solver, _ = mesh_equations_solver_cache(semi.semis[i])
+      summary_line(increment_indent(io), "mesh", mesh |> typeof |> nameof)
+      summary_line(increment_indent(io), "equations", equations |> typeof |> nameof)
+      summary_line(increment_indent(io), "initial condition", semi.semis[i].initial_condition)
+      # TODO boundary conditions? That will be 36 BCs for a cubed sphere
+      summary_line(increment_indent(io), "source terms", semi.semis[i].source_terms)
+      summary_line(increment_indent(io), "solver", solver |> typeof |> nameof)
+    end
+    summary_line(io, "total #DOFs", ndofs(semi))
+    summary_footer(io)
+  end
+end
+
+
+function summary_semidiscretization(semi::SemidiscretizationCoupled, io, io_context)
+  show(io_context, MIME"text/plain"(), semi)
+  println(io, "\n")
+  for i = 1:nsystems(semi)
+    mesh, equations, solver, _ = mesh_equations_solver_cache(semi.semis[i])
+    summary_header(io, "System #$i")
+
+    summary_line(io_context, "mesh", mesh |> typeof |> nameof)
+    show(increment_indent(io_context), MIME"text/plain"(), mesh)
+
+    summary_line(io_context, "equations", equations |> typeof |> nameof)
+    show(increment_indent(io_context), MIME"text/plain"(), equations)
+
+    summary_line(io_context, "solver", solver |> typeof |> nameof)
+    show(increment_indent(io_context), MIME"text/plain"(), solver)
+
+    summary_footer(io)
+    println(io, "\n")
+  end
+end
+
+function Base.summary(semi::SemidiscretizationCoupled)
+  _, _, solver, _ = mesh_equations_solver_cache(semi.semis[1])
+  summary(solver)
+end
+
+
+@inline Base.ndims(semi::SemidiscretizationCoupled) = ndims(semi.semis[1])
+
+@inline nsystems(semi::SemidiscretizationCoupled) = length(semi.semis)
+
+@inline Base.real(semi::SemidiscretizationCoupled) = promote_type(real.(semi.semis)...)
+
+@inline Base.eltype(semi::SemidiscretizationCoupled) = promote_type(eltype.(semi.semis)...)
+
+@inline function ndofs(semi::SemidiscretizationCoupled)
+  sum(ndofs, semi.semis)
+end
+
+@inline function nelements(semi::SemidiscretizationCoupled)
+  return sum(semi.semis) do semi_
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semi_)
+
+    nelements(mesh, solver, cache)
+  end
+end
+
+@inline function total_volume(semi::SemidiscretizationCoupled)
+  sum(semi.semis) do semi_
+    mesh, equations, solver, cache = mesh_equations_solver_cache(semi_)
+    total_volume(mesh, solver, cache)
+  end
+end
+
+
+function compute_coefficients(t, semi::SemidiscretizationCoupled)
+  @unpack u_indices = semi
+
+  u_ode = Vector{real(semi)}(undef, u_indices[end][end])
+
+  for i in 1:nsystems(semi)
+    # Call `compute_coefficients` in `src/semidiscretization/semidiscretization.jl`
+    u_ode[u_indices[i]] .= compute_coefficients(t, semi.semis[i])
+  end
+
+  return u_ode
+end
```

Comment:
@ranocha OK to resolve?

Reply:
Ok, but I still would like to add this to a TODO list for the coupled stuff. We could implement an additional layer of dispatch using the `semidiscretization` instead. My mental model is that we should have something along the lines of
```julia
function compute_coefficients(...)
  u_ode = allocate_coefficients(...)
  compute_coefficients!(u_ode, ...)
  return u_ode
end
```
since this is a very general interface and allows people to compute functions on the current mesh etc. without allocations if they have already an `u_ode`.

Reply:
Tracked in #1520.